### DISCUSSION TITLE
What is the Pros of this plugin over Logseq's native Zotero function

### DISCUSSION THREAD

**Discussion by FelixHuoEZ** [e-zz/logseq-pdf-extract#6] [Q&A]
Hello,

I recently came across this plugin and found it quite intriguing, particularly as it seems to offer functionalities similar to Logseq's built-in Zotero feature. 
Could you please elaborate on the key differences and Pros of this plugin compared to Logseq's native Zotero function?
Understanding these aspects would greatly help me in determining which tool to incorporate into my workflow.
Thanks for your dedication to developing and sharing this tool!

**Comment by e-zz** [comment]
Hi, 
At the moment, I am trying to replicate the native `/Zotero` command.  That's the first thing. So they almost do the same job. 
For a quick reference, the plugin is 
- ✔ fully local. Zotero API token is no more necessary. 
- ✔ more robust than `/Zotero`. Since it works locally, you can start to import after installation and avoid the trouble of creating a Zotero profile. In theory, it should work well even when config.edn is broken. 
    
>   But one thing in config.edn might be essential: `Zotero linked attachment base directory` or `Zotero data directory`. That's the only way to tell Logseq where to look for those PDFs managed by Zotero which are located out of the graph folder.
- ✔ faster than `/Zotero`. This plugin helps you search in your zotero library without any tangible delay, while `/Zotero` usually takes a bit of time to initiate or fetch results.
- ✔ import by selection. One feature is that you could directly import what have been selected in Zotero. To do this you can do an "empty search" in the search panel. Also, there is a shortcut `ctrl+alt+e` (Warn: only allowed in editing mode), which imports all selected items to current cursor position.
>  This comes handy with Zotero connector. Quite often we get a PDF via Zotero connector in the browser. In this case, by immediately pressing `ctrl+alt+e` in Logseq, we have the PDF imported at cursor. So within no more than 5 seconds, we could start to annotate the PDF in Logseq if we want. By comparison, using  `/Zotero` to do the same thing usually takes few steps so takes longer time, which interrupts the flow and gets quite annoying if you have to import a lot of PDFs.
- ✔ Fuzzy Search (under development). For now you could search a group of keywords from either titles, authors or tags (their combination also works). Real fuzzy search that matches the input with a part of certain word might be provided in future.
- ✔ compatible with `/Zotero` command. This plugin mimics `/Zotero` when creating an page for the imported item. It's safe to use both at the same time.
- ❌ this plugin cannot import notes. Please use `/Zotero` instead to achieve this. 
>  Currently I am engaged in what `/Zotero` can't import, like annotations in PDFs. This plugin already fetches notes from Zotero even though in the end they are not written into the page. These notes are stored as html so basically we just need to convert them into md. Hope to see PR on this soon:)

Besides, this plugin requires Zotero to be running and Zotserver installed.

So in short, you can use zotero-related feature as a full-functioning local substitution of `/Zotero`, but easier and faster most of the time. Or you could just take it as your plan B when `/Zotero` isn't working properly. 
Please let me know if any further question.

---
Though not dicrectly relevant, I think it's a good chance to mention those many possibilities that come after. For example,
- Show recent items or PDFs. I want this just because, unlike in Zotero, we cannot open multiple PDFs in Logseq. Unluckily, it's not so convenient to switch between PDFs in Logseq. Quite often I try to find the PDF I just checked one minute ago, especially when  there isn't any references to that PDF's annotations at hand.
- Customize item page. This might be helpful for special needs like https://github.com/logseq/logseq/issues/6509 
- Sync notes. In the recent update, Zotero now supports also EPUB and HTML, not just PDF. So, maybe it's a good idea to sync annotations (not notes) from Zotero. In Obsidian, a plugin called Zotilt can do this. However, it parses Zotero's database and imports users' annotations directly from there, which is impossible to try in Logseq (for safety reason I think). Now, logseq-pdf-extract could potentially provide a solution since it works in pair with a zotero plugin, ZotServer.
- For more ideas please see [#possible improvements](https://github.com/e-zz/logseq-pdf-extract?tab=readme-ov-file#possible-improvements)

The Logseq team is doing a great job and consistently bringing us amazing new features. But my feeling is that some "featuers" might not be fine-tuned for daily use. Sometimes it's possible, or even preferable, to hand over the task to users. Creating plugins like this one might inspire more users to be involed in contribution to the “Logseq ecosystem".
